owca/containers.py

```diff
@@ -53,81 +53,248 @@ def _sanitize_cgroup_path(cgroup_path):
     return relative_cgroup_path.replace('/', '-')
 
 
-@dataclass
-class Container:
-    cgroup_path: str
-    platform_cpus: int
-    resgroup: ResGroup = None
-    allocation_configuration: Optional[AllocationConfiguration] = None  # Only used for allocations.
-    rdt_enabled: bool = True
-    rdt_mb_control_enabled: bool = False
-    container_name: str = None  # defaults to flatten value of provided cgroup_path
-
-    def __post_init__(self):
-        self.cgroup = Cgroup(
-            self.cgroup_path,
-            platform_cpus=self.platform_cpus,
-            allocation_configuration=self.allocation_configuration,
-        )
-        self.container_name = (self.container_name or
-                               _sanitize_cgroup_path(self.cgroup_path))
-        self.perf_counters = PerfCounters(self.cgroup_path, event_names=DEFAULT_EVENTS)
+class ContainerInterface(ABC):
+    @abstractmethod
+    def set_resgroup(self, resgroup: ResGroup):
+        ...
+
+    @abstractmethod
+    def get_resgroup(self) -> ResGroup:
+        ...
+
+    @abstractmethod
+    def get_name(self) -> str:
+        """Returns cgroup_path cleaned."""
+
+    @abstractmethod
+    def get_cgroup(self) -> Cgroup:
+        ...
+
+    @abstractmethod
+    def get_cgroup_path(self) -> str:
+        ...
+
+    @abstractmethod
+    def get_pids(self) -> List[str]:
+        ...
+
+    @abstractmethod
+    def sync(self) -> None:
+        ...
+
+    @abstractmethod
+    def get_measurements(self) -> Measurements:
+        ...
+
+    @abstractmethod
+    def cleanup(self) -> None:
+        ...
+
+    @abstractmethod
+    def get_allocations(self) -> TaskAllocations:
+        ...
+
+
+class ContainerSet(ContainerInterface):
+    def __init__(self,
+                 cgroup_path: str, cgroup_paths: List[str], platform_cpus: int,
+                 allocation_configuration: Optional[AllocationConfiguration] = None,
```

Comment:
```suggestion
                 allocation_configuration: AllocationConfiguration = None,
```
(https://docs.python.org/3.6/library/typing.html#typing.Optional)

Reply:
Default value is None. It should stay as Optional: AllocationConfiguration or None.